src/ProgressBar/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useLayoutEffect, useRef, useState } from "react";
+
+interface ProgressBarProps {
+  /**
+   * Int from 0 to 100
+   */
+  percentComplete: number;
+  /**
+   * CSS Value; Total width of progress bar.
+   * (e.g. "50%" or "200px")
+   */
+  totalWidth?: string;
+  /**
+   * Size (height) of the ProgressBar
+   */
+  size?: "s" | "m";
+}
+
+const ProgressBar = ({
+  totalWidth = "100%",
+  percentComplete = 0,
+  size = "m",
+}: ProgressBarProps) => {
+  const [currentPct, setCurrentPct] = useState(0);
+  const lineRef = useRef<SVGLineElement>(null);
+  const animateRef = useRef<SVGAnimateElement>(null);
+
+  useLayoutEffect(() => {
+    const animFrom = percentComplete === 0 ? percentComplete : currentPct;
+    if (lineRef.current) {
+      if (animateRef.current) {
+        animateRef.current.setAttribute("from", `${animFrom}%`);
+        animateRef.current.setAttribute("to", `${percentComplete}%`);
+        animateRef.current.beginElement();
+        setCurrentPct(percentComplete);
```

Comment:
Triggers animation on the `line` element and stores the current `percentComplete` for the next render